lib/xopts.ts

```diff
@@ -0,0 +1,74 @@
+import {Signer} from 'ethers';
+import {providers} from 'ethers';
+import {ReadOnlyContracts, ReadWriteContracts} from './contracts';
+import {Addresses, Deployments} from './addresses';
+import {ExchangeRate, Currency, MonetaryAmount} from './monetary';
+
+type SignerOrProvider = Signer | providers.Provider;
+
+interface Option<From extends Currency, To extends Currency> {
+  expiry: Date;
+  from: From;
+  to: To;
+  strikePrice: ExchangeRate<From, To>;
+  currentPrice: ExchangeRate<From, To>;
+  liquidityFrom: MonetaryAmount<From>;
```

Comment:
I'm wondering if we should split the `currentPrice` and `liquidityFrom` into a separate thing. The current price and the liquidity will continuously update whereas the other properties will remain the same.

Also, if `From` is the underlying, i.e., BTC, then the liquidity would be denoted in the amount of BTC equivilent options, right?

Reply:
Yeah, we should definitely split that!